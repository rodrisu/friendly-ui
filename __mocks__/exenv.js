// Mocks the exenv library and makes its values mutable

module.exports = {
  canUseDOM: true,
  canUseEventListeners: true,
}
