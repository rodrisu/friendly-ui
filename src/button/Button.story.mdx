import { useState } from 'react'
import { action } from '@storybook/addon-actions'
import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks'

import { color } from '../_utils/branding'
import { BASIC } from '../_utils/taxonomy'
import { ArrowIcon } from '../icon/arrowIcon'
import { Button, ButtonStatus } from './index'

export const commonProps = {
  onClick: action('clicked'),
  onMouseDown: action('mouseDown'),
  onMouseUp: action('mouseUp'),
  onTouchStart: action('touchStart'),
  onTouchEnd: action('touchEnd'),
  onFocus: action('focused'),
  onBlur: action('blured'),
}

export const PrimaryButton = () => {
  const [buttonStatus, setButtonStatus] = useState(ButtonStatus.PRIMARY)
  return (
    <Button
      status={buttonStatus}
      {...commonProps}
      onClick={() => {
        setButtonStatus(ButtonStatus.LOADING)
        setTimeout(() => setButtonStatus(ButtonStatus.CHECKED), 2000)
        setTimeout(() => setButtonStatus(ButtonStatus.PRIMARY), 5000)
      }}
    >
      Click to animate
    </Button>
  )
}

<Meta title={`${BASIC}/Button`} />

# Button

## Primary button

> With text

These buttons are always centred in the page, unless it’s displayed next to a secondary button (e.g. Accept / Refuse).

When user taps on it there’s a pressed state.
After tapping if request is already loaded we either go to the next page or close the modal or if it’s a final decision we display the checked state.
If request is not loaded we display the loading state of these buttons. When loaded, we either go to the next page or close the modal or if it’s a final decision we display the checked state.

<Canvas>
  <Story name="Primary">
    <PrimaryButton />
  </Story>
</Canvas>

> With an icon

Continue buttons
These buttons are used in the middle of the flow and never at the end of it.

When user taps on it there’s a pressed state.
After tapping if request is already loaded we go to the next page.
If request is not loaded we display the loading state of these buttons. When loaded, we go to the next page. There’s no validated state for this component as it’s never a final decision button.

<Canvas>
  <Story name="Primary with icon">
    <Button
      status={ButtonStatus.PRIMARY}
      {...commonProps}
    >
      <ArrowIcon right iconColor={color.white} /> Go there
    </Button>
  </Story>
</Canvas>

### When should I use it?

> With text

Use it at the end of a flow. Allows the member to make a final decision.

- ✅ Use when making a final decision
- ✅ Always centred in the page
- ⛔️ Can’t go on 2 lines. If too long we use “…”
- ⛔️ Not always sticky
- ℹ️ No minimum size.
- 🖊 Max. 20 characters in EN, max. 30 in other languages (including spaces)

> With an icon

Use it in the middle of a flow. Allows the member to continue to the next screen.

- ✅ Always at bottom right of screen
- ✅ Always displayed when keyboard is also displayed
- ⛔️ Not always sticky.
- 🖊 No text in this component.

### Behaviour

> With Text

These buttons can either be sticky or not depending on the need.

If sticky they appear above the keyboard and above other components in the page.

When there’s an error that blocks the user to move forward, we hide these buttons.

> With an icon

These buttons are always displayed on the bottom right of the screen. They can either be sticky or not depending on the need.

If sticky they appear above the keyboard and above other components in the page.

When there’s an error that blocks the user to move forward, we hide these buttons.

### Long texts

If text is too long we can’t go on 2 lines. If too long we display “…” at the end.

## Secondary button

These buttons can be used as a secondary option when there’s already a primary button action.

When user taps on it there’s a pressed state.
After tapping if request is already loaded we either go to the next page or close the modal or if it’s a final decision we display the checked state then close the page (in some cases we don’t change the page like create alert in Search Results).
If request is not loaded we display the loading state of these buttons. When loaded, we either go to the next page or close the modal or if it’s a final decision we display the checked state.

<Canvas>
  <Story name="Secondary">
    <Button
      status={ButtonStatus.SECONDARY}
      {...commonProps}
    >
      Secondary button
    </Button>
  </Story>
</Canvas>

### When should I use it?

- ✅ Always centred in the page
- ⛔️ Can’t go on 2 lines. If too long we use “…”
- ⛔️ Not always sticky
- ℹ️ No minimum size.

### Behaviour

These buttons are always displayed on the bottom right of the screen. They can either be sticky or not depending on the need.

If sticky they appear above the keyboard and above other components in the page.

When there’s an error that blocks the user to move forward, we hide these buttons.

### Long texts

If text is too long we can’t go on 2 lines. If too long we display “…” at the end.

## Tertiary button

Least prominent button with no background and reduced padding.

When user taps on it there’s a pressed state.
After tapping if request is already loaded we either go to the next page or close the modal or if it’s a final decision we display the checked state then close the page.

If request is not loaded we display the loading state of these buttons.
When loaded, we either go to the next page or close the modal or if it’s a final decision we display the checked state.

<Canvas>
  <Story name="Tertiary">
    <Button
      status={ButtonStatus.TERTIARY}
      {...commonProps}
    >
      Tertiary button
    </Button>
  </Story>
</Canvas>

### When should I use it?

- ✅ Always centred in the page
- ℹ️ No minimum size.

### Behaviour

These buttons are used in button group for reduced component size on mobile devices.

### Long texts

If text is too long we can’t go on 2 lines. If too long we display “…” at the end.

## Variants

### Warning button
<Canvas>
  <Story name="Warning">
    <Button
      status={ButtonStatus.WARNING}
      {...commonProps}
    >
      Warning button
    </Button>
  </Story>
</Canvas>

### Facebook button
<Canvas>
  <Story name="Facebook">
    <Button
      status={ButtonStatus.FACEBOOK}
      {...commonProps}
    >
      Facebook button
    </Button>
  </Story>
</Canvas>

### VKontakte button
<Canvas>
  <Story name="Vkontakte">
    <Button
      status={ButtonStatus.VKONTAKTE}
      {...commonProps}
    >
      VKontakte button
    </Button>
  </Story>
</Canvas>

### Unstyled button
<Canvas>
  <Story name="Unstyled">
    <Button
      status={ButtonStatus.UNSTYLED}
      {...commonProps}
    >
      Unstyled button
    </Button>
  </Story>
</Canvas>

### Bubble button
<Canvas>
  <Story name="Bubble">
    <Button
      status={ButtonStatus.PRIMARY}
      isBubble
      {...commonProps}
    >
      <ArrowIcon right iconColor={color.white} />
    </Button>
  </Story>
</Canvas>

### Shadowed button
<Canvas>
  <Story name="Shadowed">
    <Button
      status={ButtonStatus.PRIMARY}
      isBubble
      shadowed
      {...commonProps}
    >
      <ArrowIcon right iconColor={color.white} />
    </Button>
  </Story>
</Canvas>

## Usage

```jsx
import { Button, ButtonStatus } from '@blablacar/ui-library/build/button'

<Button status={ButtonStatus.TERTIARY}>OK</Button>

# Button with loading/checked statuses
export const renderPrimaryButton = () => {
  const [buttonStatus, setButtonStatus] = useState(ButtonStatus.PRIMARY)
  return (
    <Button
      status={buttonStatus}
      onClick={() => {
        setButtonStatus(ButtonStatus.LOADING)
        setTimeout(() => setButtonStatus(ButtonStatus.CHECKED), 2000)
        setTimeout(() => setButtonStatus(ButtonStatus.PRIMARY), 5000)
      }}
    >
      Click to animate
    </Button>
  )
}
```

<ArgsTable of={Button} />
