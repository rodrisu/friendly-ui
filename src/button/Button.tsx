import React, { Fragment, PureComponent } from 'react'
import cc from 'classcat'
import isEmpty from 'lodash.isempty'

import { prefix } from '../_utils'
import { A11yProps, pickA11yProps } from '../_utils/interfaces'
import { Loader } from '../loader'

export enum ButtonStatus {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  TERTIARY = 'tertiary',
  WARNING = 'warning',
  FACEBOOK = 'facebook',
  VKONTAKTE = 'vkontakte',
  UNSTYLED = 'unstyled',
  LOADING = 'loading',
  CHECKED = 'checked',
}

export type ButtonProps = A11yProps &
  Readonly<{
    children: string | number | React.ReactNode
    type?: string
    href?: string | JSX.Element
    className?: string
    status?: ButtonStatus
    focus?: boolean
    isBubble?: boolean
    shadowed?: boolean
    onClick?: (event: React.MouseEvent<HTMLElement>) => void
    onBlur?: (event: React.FocusEventHandler<HTMLElement>) => void
    onFocus?: (event: React.FocusEventHandler<HTMLElement>) => void
    onMouseDown?: (event: React.MouseEvent<HTMLElement>) => void
    onMouseUp?: (event: React.MouseEvent<HTMLElement>) => void
    onTouchStart?: (event: React.TouchEvent<HTMLElement>) => void
    onTouchEnd?: (event: React.TouchEvent<HTMLElement>) => void
    onDoneAnimationEnd?: () => void
    disabled?: boolean
    index?: string
    buttonRef?: (button: HTMLButtonElement) => void
    target?: HTMLLinkElement['target']
    rel?: HTMLLinkElement['rel']
  }>

export type ButtonState = Readonly<{
  value: {
    name: string
    value: string
  }
}>

type ButtonActionEvents =
  | React.MouseEvent<HTMLElement>
  | React.TouchEvent<HTMLElement>
  | React.FocusEventHandler<HTMLElement>
type functionEvent = (event: ButtonActionEvents) => void

type TypeProps = {
  ref?: (button: HTMLButtonElement) => void
  title?: string
  href?: string
  type?: string
  onClick?: (event: React.MouseEvent<HTMLElement>) => void
  onFocus?: (event: React.FocusEventHandler<HTMLElement>) => void
  onBlur?: (event: React.FocusEventHandler<HTMLElement>) => void
  onMouseDown?: (event: React.MouseEvent<HTMLElement>) => void
  onMouseUp?: (event: React.MouseEvent<HTMLElement>) => void
  onTouchStart?: (event: React.TouchEvent<HTMLElement>) => void
  onTouchEnd?: (event: React.TouchEvent<HTMLElement>) => void
  disabled?: boolean
  index?: string
}

export const eventHandler =
  (componentEvent: functionEvent, childEvent: functionEvent) => (event: ButtonActionEvents) => {
    // eslint-disable-next-line no-unused-expressions
    componentEvent && componentEvent(event)
    // eslint-disable-next-line no-unused-expressions
    childEvent && childEvent(event)
  }

export class Button extends PureComponent<ButtonProps, ButtonState> {
  private button: HTMLButtonElement

  static defaultProps: Partial<ButtonProps> = {
    type: 'button',
    href: '',
    children: '',
    className: '',
    status: ButtonStatus.PRIMARY,
    isBubble: false,
    shadowed: false,
    focus: false,
    disabled: false,
    buttonRef() {},
  }

  componentDidMount() {
    if (this.props.focus) {
      this.button.focus()
    }
  }

  componentDidUpdate(prevProps: ButtonProps) {
    const focusChanged = prevProps.focus !== this.props.focus
    if (focusChanged && this.props.focus) {
      this.button.focus()
    }
  }

  ref = (button: HTMLButtonElement) => {
    this.button = button
    this.props.buttonRef(button)
  }

  render() {
    const {
      children,
      className,
      type,
      href,
      title,
      index,
      // Modifiers
      status,
      isBubble,
      shadowed,
      // Actions
      onClick,
      onTouchStart,
      onTouchEnd,
      onMouseDown,
      onMouseUp,
      onBlur,
      onFocus,
      onDoneAnimationEnd,
      focus,
      disabled,
      buttonRef,
      // Extend case of the button for the expand component
      ...attrs
    } = this.props

    const a11yAttrs = pickA11yProps<ButtonProps>(this.props)

    let componentType
    let typeProps: TypeProps = {}

    // If we pass a component to href, we get component type and we merge props
    if (typeof href !== 'string') {
      componentType = href.type
      typeProps = { ...href.props }
    } else if (!isEmpty(href)) {
      componentType = 'a'
      typeProps = { href }
    } else {
      componentType = 'button'
      typeProps = { type }
    }

    const hasLoader = status === ButtonStatus.LOADING || status === ButtonStatus.CHECKED

    typeProps.ref = this.ref
    typeProps.title = title
    typeProps.index = index
    typeProps.onClick = eventHandler(onClick, typeProps.onClick)
    typeProps.onMouseDown = eventHandler(onMouseDown, typeProps.onMouseDown)
    typeProps.onMouseUp = eventHandler(onMouseUp, typeProps.onMouseUp)
    typeProps.onTouchStart = eventHandler(onTouchStart, typeProps.onTouchStart)
    typeProps.onTouchEnd = eventHandler(onTouchEnd, typeProps.onTouchEnd)
    typeProps.onFocus = eventHandler(onFocus, typeProps.onFocus)
    typeProps.onBlur = eventHandler(onBlur, typeProps.onBlur)
    typeProps.disabled = hasLoader || disabled

    const props = {
      className: cc([
        prefix({ button: true }),
        prefix(
          {
            [status]: status,
            bubble: isBubble || hasLoader,
            shadowed,
          },
          'button',
        ),
        className,
      ]),
      ...typeProps,
      ...attrs,
      ...a11yAttrs,
    }

    return React.createElement(
      componentType,
      props,
      <Fragment>
        {hasLoader && (
          <Loader
            size={48}
            inline
            done={status === ButtonStatus.CHECKED}
            onDoneAnimationEnd={onDoneAnimationEnd}
          />
        )}
        {!hasLoader && children}
      </Fragment>,
    )
  }
}
