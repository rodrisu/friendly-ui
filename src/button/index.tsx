// Note: can't use styled components the same way we do for all other components,
// as buttons are created dynamically with React.createElement (and can be a "a", "button" or
// even every other React component => SlugLink for instance).
export { ButtonStatus, ButtonProps } from './Button'
export { StyledButton as Button } from './Button.style'
