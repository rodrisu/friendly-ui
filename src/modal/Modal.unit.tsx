import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import exenv from 'exenv'

import { Button } from '../button'
import { Modal, ModalSize } from './index'

describe('Modal', () => {
  it('Should have default dimmer with default props', () => {
    const wrapper = mount(<Modal />)
    expect(wrapper.find('.kirk-modal-dimmer--visible').exists()).toBe(true)
  })

  it('Should have the correct a11y attributes', () => {
    const wrapper = mount(
      <Modal isOpen aria-labelledby="labelledBy" aria-describedby="describedBy" />,
    )
    const modal = wrapper.find('[role="dialog"]')
    expect(modal.length).toBe(1)
    expect(modal.prop('aria-modal')).toBe('true')
    expect(modal.prop('aria-labelledby')).toBe('labelledBy')
    expect(modal.prop('aria-describedby')).toBe('describedBy')
  })

  it('Should be not visible if isOpen is set to false', () => {
    const wrapper = mount(<Modal />)
    expect(wrapper.find('.kirk-modal-dialog').exists()).toBe(false)
    expect(wrapper.find('.kirk-modal-dimmer--inactive').exists()).toBe(true)
  })

  it('Should be visible if isOpen is set to true', () => {
    const wrapper = mount(<Modal isOpen />)
    expect(wrapper.find('.kirk-modal-dialog').exists()).toBe(true)
    expect(wrapper.find('.kirk-modal-dimmer--active').exists()).toBe(true)
  })

  it('Could be fullscreen displayed', () => {
    const wrapper = mount(<Modal isOpen size={ModalSize.FULLSCREEN} />)
    expect(wrapper.find('.kirk-modal-dimmer--fullscreen').exists()).toBe(true)
  })

  it('Could have an hidden dimmer', () => {
    const wrapper = mount(<Modal isOpen displayDimmer={false} />)
    expect(wrapper.find('.kirk-modal-dimmer--hide').exists()).toBe(true)
    expect(wrapper.find('.kirk-modal-dimmer--visible').exists()).toBe(false)
  })

  it('Could edit the modal width', () => {
    const wrapper = mount(<Modal isOpen size={ModalSize.LARGE} />)
    expect(wrapper.find('.kirk-modal--large').exists()).toBe(true)
  })

  it('should hide the modal on close button click', () => {
    const mockClose = jest.fn()
    const wrapper = mount(<Modal isOpen onClose={mockClose} />)
    wrapper.find(Button).simulate('click')
    expect(mockClose).toHaveBeenCalledTimes(1)
  })

  it('should render the same layout on client and server side', () => {
    exenv.canUseDOM = false
    const serverSide = mount(<Modal />)
    exenv.canUseDOM = true
    const clientSide = mount(<Modal />)

    expect(serverSide.html()).toEqual(clientSide.html())
  })

  it('should not have changed', () => {
    exenv.canUseDOM = false
    const modalServerSide = renderer.create(<Modal isOpen large />).toJSON()
    exenv.canUseDOM = true
    expect(modalServerSide).toMatchSnapshot()
  })
})
