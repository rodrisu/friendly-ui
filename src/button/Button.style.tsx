import styled from 'styled-components'

import { color, componentSizes, font, shadow, space, transition } from '../_utils/branding'
import { Button } from './Button'

const borderSize = '2px'

export const StyledButton = styled(Button)`
  & {
    position: relative;
    box-sizing: border-box;
    display: inline-flex;
    padding: 0 ${space.xl};
    border: 1px solid transparent;
    border-radius: 48px;
    font-size: ${font.base.size};
    line-height: 1;
    text-align: center;
    text-decoration: none;
    cursor: pointer;
    font-family: inherit;
    height: ${componentSizes.buttonIconSize};
    min-width: ${componentSizes.buttonIconSize};
    max-width: 100%;
    overflow: hidden;
    user-select: none;
    align-items: center;
    transition: max-width ${transition.duration.fast} ease-in,
      background-color ${transition.duration.base} ease-in;
    white-space: nowrap;
  }

  & svg,
  & img {
    flex-shrink: 0;
  }

  & svg {
    margin-right: ${space.m};
  }

  &.kirk-button-bubble svg {
    margin: 0;
  }

  &.kirk-itemChoice {
    border-radius: 0;
    overflow: visible;
    width: 100%;
  }

  &:hover,
  &:active {
    outline: 0;
  }

  &.kirk-button-unstyled {
    background-color: transparent;
    border-radius: inherit;
    color: ${color.blue};
    padding: inherit;
    display: inherit;
    line-height: inherit;
    font-size: inherit;
    text-align: left;
    min-width: inherit;
    min-height: inherit;
    height: auto;
    white-space: normal;
  }

  &[disabled] {
    opacity: 0.5;
    cursor: default;
  }

  /* for unstyled disabled buttons, no opacity decrease (mainly icon buttons) */
  &.kirk-button-unstyled[disabled] {
    opacity: 1;
  }

  &.kirk-button-primary {
    background-color: ${color.blue};
    color: ${color.white};
  }

  &.kirk-button-shadowed {
    border: ${borderSize} solid ${color.white};
    box-shadow: ${shadow.default};
    box-sizing: border-box;
  }

  &.kirk-button-primary:hover,
  &.kirk-button-primary:focus,
  &.kirk-button-primary:active {
    background-color: ${color.darkBlue};
  }

  &.kirk-button-secondary {
    background-color: ${color.white};
    border-color: ${color.gray};
    color: ${color.blue};
  }
  &.kirk-button-secondary.kirk-button-shadowed {
    border: none;
  }

  &.kirk-button-secondary:hover,
  &.kirk-button-secondary:focus,
  &.kirk-button-secondary:active {
    background-color: ${color.gray};
  }

  &.kirk-button-tertiary {
    background-color: ${color.white};
    border-color: transparent;
    color: ${color.blue};
  }

  &.kirk-button-tertiary:hover,
  &.kirk-button-tertiary:focus,
  &.kirk-button-tertiary:active {
    background-color: ${color.gray};
  }

  &.kirk-button-loading {
    background-color: transparent;
    max-width: initial;
    border: 0;
  }

  &.kirk-button-loading:hover,
  &.kirk-button-loading:focus,
  &.kirk-button-loading:active {
    background-color: transparent;
  }

  &.kirk-button-loading[disabled],
  &.kirk-button-checked[disabled] {
    opacity: 1;
  }

  & svg.kirk-button-loader {
    position: absolute;
    top: 0;
    left: 50%;
    margin-left: -24px;
  }

  &.kirk-button-checked {
    border: 0;
  }

  &.kirk-button-warning {
    background-color: ${color.red};
    color: ${color.white};
  }

  &.kirk-button-bubble {
    display: inline-block;
    text-align: center;
    padding: 0;
    line-height: 0;
    width: ${componentSizes.buttonIconSize};
    height: ${componentSizes.buttonIconSize};
  }

  &.kirk-button-bubble.kirk-button-shadowed {
    width: calc(${componentSizes.buttonIconSize} - ${borderSize} * 2);
    height: calc(${componentSizes.buttonIconSize} - ${borderSize} * 2);
  }

  &.kirk-button-facebook {
    background-color: ${color.facebookBrand};
    color: ${color.white};
  }

  &.kirk-button-vkontakte {
    background-color: ${color.vkBrand};
    color: ${color.white};
  }

  @media (hover: none), (hover: on-demand) {
    &.kirk-button-secondary:hover {
      background-color: ${color.white};
    }
  }
`
