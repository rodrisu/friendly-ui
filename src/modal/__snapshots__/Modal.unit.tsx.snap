// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should not have changed 1`] = `
.c3 {
  fill: #00AFF5;
}

.c3.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c3.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c3 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c2 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0 24px;
  border: 1px solid transparent;
  border-radius: 48px;
  font-size: 16px;
  line-height: 1;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  font-family: inherit;
  height: 48px;
  min-width: 48px;
  max-width: 100%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: max-width 500ms ease-in, background-color 200ms ease-in;
  transition: max-width 500ms ease-in, background-color 200ms ease-in;
  white-space: nowrap;
}

.c2 svg,
.c2 img {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2 svg {
  margin-right: 8px;
}

.c2.kirk-button-bubble svg {
  margin: 0;
}

.c2.kirk-itemChoice {
  border-radius: 0;
  overflow: visible;
  width: 100%;
}

.c2:hover,
.c2:active {
  outline: 0;
}

.c2.kirk-button-unstyled {
  background-color: transparent;
  border-radius: inherit;
  color: #00AFF5;
  padding: inherit;
  display: inherit;
  line-height: inherit;
  font-size: inherit;
  text-align: left;
  min-width: inherit;
  min-height: inherit;
  height: auto;
  white-space: normal;
}

.c2[disabled] {
  opacity: 0.5;
  cursor: default;
}

.c2.kirk-button-unstyled[disabled] {
  opacity: 1;
}

.c2.kirk-button-primary {
  background-color: #00AFF5;
  color: #FFF;
}

.c2.kirk-button-shadowed {
  border: 2px solid #FFF;
  box-shadow: 0 2px 4px rgba(0,0,0,.3);
  box-sizing: border-box;
}

.c2.kirk-button-primary:hover,
.c2.kirk-button-primary:focus,
.c2.kirk-button-primary:active {
  background-color: #008FC1;
}

.c2.kirk-button-secondary {
  background-color: #FFF;
  border-color: #DDD;
  color: #00AFF5;
}

.c2.kirk-button-secondary.kirk-button-shadowed {
  border: none;
}

.c2.kirk-button-secondary:hover,
.c2.kirk-button-secondary:focus,
.c2.kirk-button-secondary:active {
  background-color: #DDD;
}

.c2.kirk-button-tertiary {
  background-color: #FFF;
  border-color: transparent;
  color: #00AFF5;
}

.c2.kirk-button-tertiary:hover,
.c2.kirk-button-tertiary:focus,
.c2.kirk-button-tertiary:active {
  background-color: #DDD;
}

.c2.kirk-button-loading {
  background-color: transparent;
  max-width: initial;
  border: 0;
}

.c2.kirk-button-loading:hover,
.c2.kirk-button-loading:focus,
.c2.kirk-button-loading:active {
  background-color: transparent;
}

.c2.kirk-button-loading[disabled],
.c2.kirk-button-checked[disabled] {
  opacity: 1;
}

.c2 svg.kirk-button-loader {
  position: absolute;
  top: 0;
  left: 50%;
  margin-left: -24px;
}

.c2.kirk-button-checked {
  border: 0;
}

.c2.kirk-button-warning {
  background-color: #F53F5B;
  color: #FFF;
}

.c2.kirk-button-bubble {
  display: inline-block;
  text-align: center;
  padding: 0;
  line-height: 0;
  width: 48px;
  height: 48px;
}

.c2.kirk-button-bubble.kirk-button-shadowed {
  width: calc(48px - 2px * 2);
  height: calc(48px - 2px * 2);
}

.c2.kirk-button-facebook {
  background-color: #1877F2;
  color: #FFF;
}

.c2.kirk-button-vkontakte {
  background-color: #2787F5;
  color: #FFF;
}

.c1.fade {
  -webkit-transition: ease-in-out 200ms;
  transition: ease-in-out 200ms;
  -webkit-transition-property: opacity;
  transition-property: opacity;
}

.c1.fade-entering {
  opacity: 0;
}

.c1.fade-entered {
  opacity: 1;
}

.c1.fade-exiting {
  opacity: 0;
}

.c1.slide-up {
  -webkit-transition: ease-in-out 200ms;
  transition: ease-in-out 200ms;
  -webkit-transition-property: opacity,-webkit-transform;
  -webkit-transition-property: opacity,transform;
  transition-property: opacity,transform;
}

.c1.slide-up-entering {
  opacity: 0;
  -webkit-transform: translateY(50%);
  -ms-transform: translateY(50%);
  transform: translateY(50%);
}

.c1.slide-up-entered {
  opacity: 1;
}

.c1.slide-up-exiting {
  opacity: 0;
  -webkit-transform: translateY(50%);
  -ms-transform: translateY(50%);
  transform: translateY(50%);
}

.c0.kirk-modal-dimmer,
.c0.kirk-modal-dimmer--fullscreen {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  overflow: hidden;
  z-index: 999;
}

.c0.kirk-modal-dimmer--visible {
  background-color: rgba(0,0,0,0.5);
}

.c0.kirk-modal-dimmer--hide {
  background-color: #FFF;
}

.c0.kirk-modal-dimmer--fullscreen {
  background-color: transparent;
}

.c0.kirk-modal-dimmer--inactive {
  background-color: transparent;
  pointer-events: none;
}

.c0 .kirk-modal {
  position: fixed;
  top: 0;
  right: 0;
  width: 100%;
  height: 100%;
  overflow-y: auto;
}

.c0 .kirk-modal-dialog {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
  margin: 24px auto;
  width: 662px;
  background-color: #FFF;
  border-radius: 8px;
}

.c0 .kirk-modal--small .kirk-modal-dialog {
  width: 400px;
}

.c0 .kirk-modal--large .kirk-modal-dialog {
  width: 928px;
}

.c0 .kirk-modal-body {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0.kirk-modal-dimmer--fullscreen .kirk-modal-dialog {
  margin: 0;
  border-radius: 0;
  box-shadow: none;
  width: auto;
  min-height: 100%;
}

.c0 .kirk-modal--hasCloseButton .kirk-modal-dialog {
  padding-top: 70px;
}

.c0 .kirk-modal--hasCloseButton .kirk-modal-closeButton {
  position: absolute;
  top: 8px;
  left: 8px;
  background-color: transparent;
}

@media (hover:none),(hover:on-demand) {
  .c2.kirk-button-secondary:hover {
    background-color: #FFF;
  }
}

@media (max-width:799px) {
  .c0 .media-section {
    margin-left: -24px;
    margin-right: -24px;
  }
}

<div
  className="c0 kirk-modal-dimmer kirk-modal-dimmer--visible kirk-modal-dimmer--active"
>
  <div
    className="transition-wrapper"
  >
    <div
      aria-modal="true"
      className="c1 kirk-modal kirk-modal--medium kirk-modal--hasCloseButton slide-up slide-up-entered"
      onScroll={[Function]}
      role="dialog"
    >
      <div
        className="kirk-modal-dialog"
      >
        <button
          className="kirk-button kirk-button-primary kirk-button-bubble c2 kirk-modal-closeButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="kirk-icon c3"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            
            <path
              d="M19 5L5 19M19 19L5 5"
              fill="none"
              stroke="#00AFF5"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeMiterlimit="10"
              strokeWidth="2"
            />
          </svg>
        </button>
        <div
          className="kirk-modal-body"
        />
      </div>
    </div>
  </div>
</div>
`;
